### PR TITLE
Add support for getting a user from intercom

### PR DESCRIPTION
**Pull request by click-raduchiorean**

### CODE REVIEW

File: cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java [hunk 1]
```diff
@@ -188,28 +191,65 @@ public void sendMetric(final Metric metric) {
...
 
         if (!Equals.isNullOrBlank(user.emailAddress())) {
             intercomUser.setEmail(user.emailAddress());
         }
 
-        if (!Equals.isNullOrBlank(user.organisationId())) {
-            final Company company = new Company();
-            company.setCompanyID(user.organisationId());
-            intercomUser.addCompany(company);
+        if (!user.organisationIds().isEmpty()) {
```
Comment (clicktravel-steffan) [inline cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java]: This check is redundant. If the collection is empty, the `forEach` below will not have any effect.

File: cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java [hunk 2]
```diff
@@ -188,28 +191,65 @@ public void sendMetric(final Metric metric) {
...
+                company.setCompanyID(organisationId);
+                intercomUser.addCompany(company);
+            });
         }
 
         return intercomUser;
     }
 
+    private Map<String, Object> getCustomAttributes(final User user) {
+        final Map<String, ?> customAttributes = user.getCustomAttributes();
```
Comment (clicktravel-steffan) [inline cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java]: The type of this should be `Map<String, CustomAttribute<?>>`

File: cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java [hunk 3]
```diff
@@ -188,28 +191,65 @@ public void sendMetric(final Metric metric) {
...
+        return customAttributes.entrySet().stream()
+                .collect(Collectors.toMap(entry -> entry.getKey(), entry -> (Object) entry.getValue()));
```
Comment (clicktravel-steffan) [inline cheddar/cheddar-metrics-intercom/src/main/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollector.java]: The cast to `Object` should not be here

**Comment by clicktravel-chrishern** [comment]
Merging this PR now.  Will fix Steffan's comment on another PR.

**Pull request merged**